presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergAbstractMetadata.java

```diff
@@ -166,8 +169,13 @@ public IcebergAbstractMetadata(
         this.rowExpressionService = requireNonNull(rowExpressionService, "rowExpressionService is null");
         this.nodeVersion = requireNonNull(nodeVersion, "nodeVersion is null");
     }
-
-    protected abstract Table getIcebergTable(ConnectorSession session, SchemaTableName schemaTableName);
+    protected final Table getIcebergTable(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        return icebergTables.computeIfAbsent(
+                schemaTableName,
+                ignored -> getRawIcebergTable(session, schemaTableName));
+    }
+    protected abstract Table getRawIcebergTable(ConnectorSession session, SchemaTableName schemaTableName);
```

Comment:
nit: provide some javadocs to instruct users towards `getIcebergTable` instead of this method